peps/pep-0819.rst

```diff
@@ -0,0 +1,347 @@
+Post-History: Pending
+
+
+Abstract
+========
+
+Python package metadata ("core metadata") was first defined in :pep:`241` to
+use :rfc:`822` email headers to encode information about packages. This was
+reasonable at the time; email messages were the only widely used, standardized
+text format that had a parser in the standard library at the time. However,
+issues with handling different encodings, differing handling of line breaks,
+and other differences between implementations have caused numerous packaging
+bugs. To resolve these issues, this PEP proposes introducing
+`Javascript Object Notation (JSON) <https://www.json.org/json-en.html>`_
+encoded core metadata and wheel file format metadata files in Python packages.
+
+
+Motivation
+==========
+
+The email message format has a number of complexities and limitations which
+reduce its utility as a portable textual interchange format for packaging
+metadata. Due to the :mod:`email` parser requiring configuration changes to
+properly generate valid core metadata, many projects do not use the
+:mod:`!email` module and instead generate core metadata in a custom manner.
+There are many pitfalls with generating email headers that these custom
+generators can hit. First, core metadata fields may contain newlines in the
+value of fields. These newlines must be handled properly to "unfolded" multiple
+lines per :rfc:`822`. One particularly difficult to encode field is the
+``Description`` field, which may contain newlines and indentation. To encode
+the field in email headers, CRLF line breaks must be followed by seven (7)
+spaces and a pip ("|") character. While ``Description`` may now be encoded in
+the message body, similar escaping issues occur for the ``Author`` and
+``Maintainer`` fields. Improperly escaped newlines can lead to missing,
+partial, or invalid core metadata. Second, as discussed in the
+`core metadata specifications <https://packaging.python.org/specifications/core-metadata/>`__:
+
+.. epigraph::
+    The standard file format for metadata (including in wheels and installed
+    projects) is based on the format of email headers. However, email formats
+    have been revised several times, and exactly which email RFC applies to
+    packaging metadata is not specified. In the absence of a precise
+    definition, the practical standard is set by what the standard library
+    :mod:`email.parser` module can parse using the
+    :data:`email.policy.compat32` policy.
+
+Since no specific email RFC is selected, the current core metadata
+specification is ambiguous whether a given core metadata document is valid.
+:rfc:`822` is the only email standard to be explicitly listed in a PEP.
+However, the core metadata specifications also requires that core metadata is
+encoded using UTF-8 when written to a file. This de-facto makes the core
+metadata follow :rfc:`6532`, which specifies internationalization of email
+headers. This has practical interoperability concerns. Until a few years ago,
```

Comment:
Could we be more precicse here?

Reply:
Oh perhaps you mean "a few years ago". In which case I believe this would be the proper change.
```suggestion
headers. This has practical interoperability concerns. Until three years ago,
```
(ref https://github.com/pypa/packaging.python.org/pull/1046)